t/unit/backends/test_s3.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import absolute_import, unicode_literals
+
+from case import patch, MagicMock
```

Comment:
The unit tests seem fine, but apart from that, we have used [moto](https://github.com/spulec/moto) in the past, to add some functional tests. You could check the DynamoDB backend code.

Reply:
@georgepsarakis thank you, i didn't knew this program and it looks nice :)
will do it and push again soon